ksql-engine/src/test/resources/query-validation-tests/extract-json-field.json

```diff
@@ -22,7 +22,7 @@
       "name": "extract JSON array field",
       "statements": [
         "CREATE STREAM TEST (array_field ARRAY<VARCHAR>) WITH (kafka_topic='test_topic', value_format='JSON');",
-        "CREATE STREAM OUTPUT AS SELECT EXTRACTJSONFIELD(array_field[0], '$.nested') AS Col1, EXTRACTJSONFIELD(array_field[1], '$.nested') AS Col2 FROM TEST;"
+        "CREATE STREAM OUTPUT AS SELECT EXTRACTJSONFIELD(array_field[1], '$.nested') AS Col1, EXTRACTJSONFIELD(array_field[2], '$.nested') AS Col2 FROM TEST;"
```

Comment:
Add tests that test both old style and new style array access (with appropriate properties set), and one proving the default is the old base-zero. (See substring.json for examples).

Reply:
Added the tests.